demo_nodes_cpp_native/src/talker.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+
+#include "std_msgs/msg/string.hpp"
+
+#include "rmw_fastrtps_cpp/get_participant.hpp"
+#include "rmw_fastrtps_cpp/get_publisher.hpp"
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::node::Node::make_shared("talker_native");
+
+  {
+    rcl_node_t * rcl_node = node->get_node_base_interface()->get_rcl_node_handle();
+    rmw_node_t * rmw_node = rcl_node_get_rmw_handle(rcl_node);
+    eprosima::fastrtps::Participant * p = rmw_fastrtps_cpp::get_participant(rmw_node);
+    printf("eprosima::fastrtps::Participant * %zu\n", reinterpret_cast<size_t>(p));
```

Comment:
Does this actually prove anything? Maybe it would be better to call a function on the participant, something like getting the name out of it or the entity id? Then at least you know the pointer is valid.